readthedocs/api/v2/views/footer_views.py

```diff
@@ -63,113 +64,149 @@ def get_version_compare_data(project, base_version=None):
-    subproject = request.GET.get('subproject', False)
-    source_suffix = request.GET.get('source_suffix', '.rst')
-
-    # Hack in a fix for missing version slug deploy that went out a while back
-    if version_slug == '':
-        version_slug = LATEST
-
-    new_theme = (theme == 'sphinx_rtd_theme')
-    using_theme = (theme == 'default')
-    project = get_object_or_404(Project, slug=project_slug)
-    version = get_object_or_404(
-        Version.objects.public(
-            request.user,
-            project=project,
-            only_active=False,
-        ),
-        slug__iexact=version_slug,
-    )
-    main_project = project.main_language_project or project
 
-    if page_slug and page_slug != 'index':
-        if main_project.documentation_type == 'sphinx_htmldir':
-            path = page_slug + '/'
-        else:
-            path = page_slug + '.html'
-    else:
-        path = ''
-
-    version_compare_data = get_version_compare_data(project, version)
-
-    context = {
-        'project': project,
-        'version': version,
-        'path': path,
-        'downloads': version.get_downloads(pretty=True),
-        'current_version': version.verbose_name,
-        'versions': project.ordered_active_versions(user=request.user),
-        'main_project': main_project,
-        'translations': main_project.translations.all(),
-        'current_language': project.language,
-        'using_theme': using_theme,
-        'new_theme': new_theme,
-        'settings': settings,
-        'subproject': subproject,
-        'github_edit_url': version.get_github_url(
-            docroot,
-            page_slug,
-            source_suffix,
-            'edit',
-        ),
-        'github_view_url': version.get_github_url(
-            docroot,
-            page_slug,
-            source_suffix,
-            'view',
-        ),
-        'gitlab_edit_url': version.get_gitlab_url(
-            docroot,
-            page_slug,
-            source_suffix,
-            'edit',
-        ),
-        'gitlab_view_url': version.get_gitlab_url(
-            docroot,
-            page_slug,
-            source_suffix,
-            'view',
-        ),
-        'bitbucket_url': version.get_bitbucket_url(
-            docroot,
-            page_slug,
-            source_suffix,
-        ),
-        'theme': theme,
-    }
+    http_method_names = ['get']
+    permission_classes = [AllowAny]
+    renderer_classes = [JSONRenderer, JSONPRenderer]
 
-    html = template_loader.get_template('restapi/footer.html').render(
-        context,
-        request,
-    )
-    resp_data = {
-        'html': html,
-        'show_version_warning': project.show_version_warning,
-        'version_active': version.active,
-        'version_compare': version_compare_data,
-        'version_supported': version.supported,
-    }
+    def _get_project(self):
+        cache_key = '_cached_project'
+        project = getattr(self, cache_key, None)
```

Comment:
Fancy cache system, because `_get_project` and `_get_version` are called many times :)

Reply:
This should be noted in the docstring, not a PR comment ;)

Basically any time you want to write a PR comment explaining the code, put it in a code comment instead, so it stays with the code.